debian/rules

```diff
@@ -47,6 +47,10 @@ export AUTHD_SKIP_ROOT_TESTS := 1
 export DH_GOLANG_BUILDPKG := $(AUTHD_GO_PACKAGE)/... \
 	$(NULL)
 
+# Add versioned go version to the path
+AUTHD_GO_VERSION := $(shell grep-dctrl -s Build-Depends -n - ./debian/control | sed -n "s,.*\bgolang-\([0-9.]\+\)\b.*,\1,p")
```

Comment:
I’m unsure this work: if golang-go in p is 1.24 and there is no 1.23 package, we will still match 1.23 that we try to add to the path but doesn’t exist.
So, I see that you are now cheating by adding PATH as first :) So, that’s fine. I think the variable name is misleading though.

What about `AUTHD_BACKPORTED_GO_VERSION` and add a comment that it may not exists in the distro, and if golang-go is installed, this is what takes precedences?

Reply:
Fair, I've added some more verbose comment and changed the name, indeed the `$PATH` hack order isn't the perfect one but I didn't want to end up having a proper script to handle this, but this should be already enough to pick the first good version that we've installed.

Reply:
No worry! I think this is good and now can’t lead to misunderstanding of the logic.